src/Table.hxx

```diff
@@ -166,13 +224,26 @@ public:
         write_sql_inserts(os, table_name);
     }
     void write_sqlite_db(const boost::filesystem::path &path) const;
-    void write_fits(std::ostream &os) const;
-    void write_fits(const boost::filesystem::path &filename) const;
-    void write_fits(fitsfile *fits_file) const;
-    void write_tabledata(std::ostream &os, const Format::Enums &output_format) const;
+    void write_fits(
+            std::ostream &os,
+            const std::map<size_t, Data_Type> &datatypes_for_writing) const;
+    void write_fits(
+            const boost::filesystem::path &filename,
+            const std::map<size_t, Data_Type> &datatypes_for_writing) const;
+    void write_fits(
+            fitsfile *fits_file,
+            const std::map<size_t, Data_Type> &datatypes_for_writing) const;
+    void write_tabledata(std::ostream &os, const Format::Enums &output_format,
+                         const std::map<size_t, Data_Type> &datatypes) const;
+
+    void write_tabledata(std::ostream &os, const Format::Enums &output_format) const {
+         write_tabledata(os, output_format, std::map<size_t, Data_Type>());
+    }
+
     void write_html(std::ostream &os) const;
     boost::property_tree::ptree generate_property_tree(
-            const std::string &tabledata_string) const;
+            const std::string &tabledata_string,
+            const std::map<size_t, Data_Type> &alternative_datatypes) const;
```

Comment:
I'm probably missing something, but isn't letting users pass in `alternative_datatypes` in the various write methods (versus computing appropriate output types internally, in a non-publicly accessible way) a foot-gun?

Reply:
Good point - you convinced me we should leave it as is, and come back to it in a more principled way later.